CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/parser/PCAPImportFileParser.java

```diff
@@ -0,0 +1,173 @@
+            
+            // Open the supplied PCAP file
+            final Pcap pcap = Pcap.openStream(input.getFile());
+            frameCounter = 1;
+            
+            pcap.loop(new PacketHandler() {
+                @Override
+                public boolean nextPacket(Packet packet) throws IOException {
+                    
+                    IPPacket ipPacket = null;
+                    
+                    SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+                    Date timestamp = new Date(packet.getArrivalTime() / 1000);
+                    String frame = Integer.toString(frameCounter);
+                    String sourceIP = "";
+                    String sourcePort = "";
+                    String sourceType = "";
+                    String destinationIP = "";
+                    String destinationPort = "";
+                    String destinationType = "";
+                    String protocol = "";
+                    String ipVersion = "";
+                    Buffer buffer = packet.getPayload();
+                    String length = Integer.toString(buffer.capacity());
+                         
+                    try {               
+                        
+                        if (packet.hasProtocol(Protocol.TCP)) {
+                            protocol = "TCP";
+                            TCPPacket tcpPacket = (TCPPacket) packet.getPacket(Protocol.TCP);
+                            sourcePort = Integer.toString(tcpPacket.getSourcePort());
+                            destinationPort = Integer.toString(tcpPacket.getDestinationPort());
+                            ipPacket = (IPPacket) tcpPacket.getParentPacket();
+                        } else if (packet.hasProtocol(Protocol.UDP)) {
+                            protocol = "UDP";
+                            UDPPacket udpPacket = (UDPPacket) packet.getPacket(Protocol.UDP);
+                            sourcePort = Integer.toString(udpPacket.getSourcePort());
+                            destinationPort = Integer.toString(udpPacket.getDestinationPort());
+                            ipPacket = (IPPacket) udpPacket.getParentPacket();
+                        }
+                        
+                        if (ipPacket != null) {
+                            ipVersion = "IPv" + Integer.toString(ipPacket.getVersion());
+                            sourceIP = ipPacket.getSourceIP();
+                            sourceType = ipVersion + " Address";
+                            destinationIP = ipPacket.getDestinationIP();
+                            destinationType = ipVersion + " Address";
+                        }
+                        
+                        LOGGER.log(Level.INFO, "{0}, {1}, {2}, {3}, {4}, {5}, {6}, {7}, {8}, {9}, {10}", new Object[]{frame, formatter.format(timestamp), sourceIP, sourcePort, sourceType, destinationIP, destinationPort, destinationType, protocol, length });
```

Comment:
Split these long lines across multiple lines for readability.

Reply:
It also appears you have 11 arguments but only 10 values

Reply:
Log message is only there for intial test purposes as this was a prototype, much easier to remove the line which achieves nothing.